src/main/java/net/pms/network/webguiserver/servlets/SettingsApiServlet.java

```diff
@@ -359,4 +361,37 @@ public static String getConfigurationUpdate(String key) {
 		}
 		return "";
 	}
+
+	/**
+	 * Note: This is not guaranteed to contain ALL settings,
+	 * only the ones the user has changed from defaults. To
+	 * get the whole picture it needs to be combined with
+	 * the defaults.
+	 *
+	 * Note: We do not save the configuration as JSON at
+	 * any point, this is just a convenience method for
+	 * our REST API.
+	 *
+	 * @return the user settings as a JSON string.
+	 */
+	public String getConfigurationAsJsonString() {
+		Properties configurationAsProperties = ConfigurationConverter.getProperties(CONFIGURATION.getConfiguration());
+
+		Map<String, String> propsAsStringMap = new HashMap<>();
+		configurationAsProperties.forEach((key, value) -> {
+				String strKey = Objects.toString(key);
+				if (SettingsApiServlet.haveKey(strKey)) {
+					String strValue = Objects.toString(value);
+					//do not add non acceptable empty key then it back to default
+					if (StringUtils.isNotEmpty(strValue) || SettingsApiServlet.acceptEmptyValueForKey(strKey)) {
+						//escape "\" char with "\\" otherwise json will fail
+						propsAsStringMap.put(strKey, strValue.replace("\\", "\\\\"));
+					}
+				}
+			}
+		);
```

Comment:
```suggestion
		configurationAsProperties.forEach((key, value) -> {
			String strKey = Objects.toString(key);
			if (SettingsApiServlet.haveKey(strKey)) {
				String strValue = Objects.toString(value);
				//do not add non acceptable empty key then it back to default
				if (StringUtils.isNotEmpty(strValue) || SettingsApiServlet.acceptEmptyValueForKey(strKey)) {
					//escape "\" char with "\\" otherwise json will fail
					propsAsStringMap.put(strKey, strValue.replace("\\", "\\\\"));
				}
			}
		});
```